scripts/hud/boardrig.lua

```diff
@@ -328,14 +330,131 @@ end
 
 -- ===
 
+local SPRINT_HILITE_CLR = { 142/255, 247/255, 142/255, 1 } -- 247, 247, 142 = sprint colour
+
+local function UITRpreviewSprintNoise(boardrig, unit, path)
+    local pathCells = util.tcopy(path)
+    local sim = boardrig:getSim()
+    local sprintNoise = unit:getTraits().dashSoundRange
+    -- only show for sprinting units, and only if we know the path
+    if not unit or unit:getTraits().sneaking or not pathCells then
+        return
+    end
+    -- only show if distance is sprintable
+    if not boardrig._game.hud._bValidMovement then
+        return
+    end
+    -- only show if in tactical view; that'd need to be coupled with tac view toggle amendments
+    --[[if not boardrig._game:getGfxOptions().bTacticalView then
+        return
+    end]]
+
+    -- remove cell we're at; we won't make noise from it
+    local coords = table.remove(pathCells, 1)
+    local originCell = sim:getCell(coords.x, coords.y)
+
+    -- for each tile in path, add known tiles in a radius (r = sprintNoise) to a list 
+    local cells, units = {}, {}
+
+    for i, cellCoords in pairs(pathCells) do
+    --local cellCoords = table.remove(pathCells)
+        -- not using simquery.fillCircle directly, to avoid over-iteration
```

Comment:
This shouldn't follow `simquery.fillCircle` behavior in the first place. Range is checked by the listener (`Senses:processSoundTrigger` at `sim/btree/senses.lua:535`, `SoundBug:onTrigger` at `sim/units/simsoundbug.lua:86`).

I believe the only difference is that we shouldn't take the `floor` of the distance; the decimal value is checked directly.